presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/GeoFunctions.java

```diff
@@ -378,6 +388,100 @@ public static Slice stCentroid(@SqlType(GEOMETRY_TYPE_NAME) Slice input)
         return serialize(geometry.getCentroid());
     }
 
+    @Description("Returns the Point value that is the mathematical centroid of a Spherical Geography")
+    @ScalarFunction("ST_Centroid")
+    @SqlType(SPHERICAL_GEOGRAPHY_TYPE_NAME)
+    public static Slice stSphericalCentroid(@SqlType(SPHERICAL_GEOGRAPHY_TYPE_NAME) Slice input)
+    {
+        OGCGeometry geometry = EsriGeometrySerde.deserialize(input);
+        if (geometry.isEmpty()) {
+            return null;
+        }
+        // TODO: add support for other types e.g. POLYGON
+        validateSphericalType("ST_Centroid", geometry, EnumSet.of(POINT, MULTI_POINT));
+        if (! (geometry instanceof OGCGeometryCollection)) {
+            return input;
+        }
+
+        OGCGeometryCollection geometryCollection = (OGCGeometryCollection)geometry;
+        for (int i = 0; i < geometryCollection.numGeometries(); i++) {
+            OGCGeometry g = geometryCollection.geometryN(i);
+            validateSphericalType("ST_Centroid", g, EnumSet.of(POINT));
+            Point p = (Point) g.getEsriGeometry();
+            checkLatitude(p.getX());
+            checkLongitude(p.getY());
+        }
+
+        Point centroid = null;
+        if (geometryCollection.numGeometries() == 1) {
+            return input;
+        }else if(geometryCollection.numGeometries() == 2) {
+            // 2 points
+            CartesianPoint cp1 = new CartesianPoint((Point)geometryCollection.geometryN(0).getEsriGeometry());
+            CartesianPoint cp2 = new CartesianPoint((Point)geometryCollection.geometryN(1).getEsriGeometry());
+            centroid = CartesianPoint.convertToLatLong(
```

Comment:
Right the assumption is that since we have a sphere,  the centroid of the chord between cp1 and cp2 has the same (long, lat) as the centroid on the sphere.